### ISSUE TITLE
Builder.from() does not honor collection nullability

### ISSUE THREAD

**Issue by nebhale** [immutables/immutables#339]
Given the following class:

```
@Value.Immutable
abstract class FromNullCollection {
    @Nullable abstract List<String> getItems();
}
```

the following `Builder` methods are generated:

```
public final Builder from(FromNullCollection instance) {
    Objects.requireNonNull(instance, "instance");
    addAllItems(instance.getItems());
    return this;
}
```

```
public final Builder addAllItems(Iterable<String> elements) {
    Objects.requireNonNull(elements, "items element");
    if (this.items == null) {
        this.items = new ArrayList<String>();
    }
    for (String element : elements) {
        this.items.add(Objects.requireNonNull(element, "items element"));
    }
    return this;
}
```

```
public final Builder items(@Nullable Iterable<String> elements) {
    if (elements == null) {
        this.items = null;
        return this;
    }
    this.items = new ArrayList<String>();
    return addAllItems(elements);
}
```

This code, generated by 2.2 creates the `items()` and associated builder methods properly honoring nullability.  However, the `from()` method assumes that the value returned from the getter is never `null`, passing it to `addAllItems()` which enforces non-`null`.

**Comment by elucash** [comment]
In this case `items` should be called. Thank you for reporting this!

**Comment by nebhale** [comment]
My pleasure.  Sorry for not getting to review the nullable collections stuff before 2.2 made it out, but this is a pretty minor bug from our perspective.  We're really happy with there results so far.

**Comment by elucash** [comment]
Just thought about it again, as the nature of `from` method (which is short for `mergeFrom` or `copyFrom`) is additive, here probably it's better to not call `items`, but rather call `addAllItems` if checked to be non-null.

Thank you for feedback! We are planning a bugfix release in 2 days.

**Comment by nebhale** [comment]
Just reviewed this change in the latest `SNAPSHOT` and it looks good.

**Comment by elucash** [comment]
Released in 2.2.1 Thank you!
